lib/LibError.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+# include <exception>
+
+class   LibError: public std::exception
```

Comment:
```suggestion
class LibError
```
Inheriting from std::exception is useless. Don't forget to remove the include.